sys/shell/shell.c

```diff
@@ -298,12 +346,22 @@ void shell_run_once(const shell_command_t *shell_commands,
     while (1) {
         int res = readline(line_buf, len);
 
-        if (res == EOF) {
-            break;
-        }
+        switch (res) {
 
-        if (!res) {
-            handle_input_line(shell_commands, line_buf);
+            case EOF:
+                return;
+
+            case ERROR_READLINE_EXCEEDED:
+                puts("shell: maximum line length exceeded");
```

Comment:
What usually happens on a terminal when the input max is exceeded?? e.g.:

```
$ /bin/echo * > /dev/null
bash: /bin/echo: Argument list too long
```

@kaspar030 do you have some though on this?